src/query/interpreter.cpp

```diff
@@ -1818,7 +1818,7 @@ PreparedQuery PrepareProfileQuery(ParsedQuery parsed_query, bool in_explicit_tra
 PreparedQuery PrepareDumpQuery(ParsedQuery parsed_query, CurrentDB &current_db) {
   MG_ASSERT(current_db.execution_db_accessor_, "Dump query expects a current DB transaction");
   auto *dba = &*current_db.execution_db_accessor_;
-  auto plan = std::make_shared<PullPlanDump>(dba);
+  auto plan = std::make_shared<PullPlanDump>(dba, current_db.db_acc_->get()->trigger_store());
```

Comment:
Unchecked access to db_acc. AFAIK, db_acc should always have value but again, someone may change the code and we are in troubles. Ask @andrejtonev if db_acc can ever be without value. And if no, can we remove std::optional

Reply:
Quick background:
The `memgraph::dbms::DatabaseAccess` is a mechanism used to make sure a database exists. The user can CREATE/DROP databases at any point. The access is a reference counter that protects a database from being dropped while used by someone/something else.

At this point the `db_acc_` has to have a value. The problem is that the plan will be executed in the future and we don't control when that is.
What you should do is copy the `DatabaseAccess` itself into the plan and call `db_acc->trigger_store()` from the plan. This way the database cannot be dropped until the plan has been executed.